### PR TITLE
nextcloud: fix deprecation warning

### PR DESCRIPTION
**Pull request by averelld**
###### Motivation for this change
#65706 deprecates phpfpm poolConfigs, but the nextcloud module still sets those by default. This is supposed to get rid of that deprecation warning. Could use some independent testing.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @aanderse

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg test nextcloud.basic

File: nixos/modules/services/web-apps/nextcloud.nix [hunk 1]
```diff
@@ -287,6 +295,11 @@ in {
           message = "Please specify exactly one of adminpass or adminpassFile";
         }
       ];
+
+      warnings = optional (cfg.poolConfig != null) ''
```
Comment (aanderse) [inline nixos/modules/services/web-apps/nextcloud.nix]: @averelld your intention with this was to present the user with a second warning to clarify the first warning I assume?

**Review by aanderse — APPROVED** [review]
I haven't tested, but overall LGTM :+1:
I assume all the tests (aside from the `redis` test mentioned by @jonringer) pass now?

**Comment by jonringer** [comment]
I tested locally, and the other two pass

**Review by Ma27 — APPROVED** [review]
Diff seems fine, also tested this on my personal Nextcloud instance. The broken test gets fixed in #68441, so this should be safe to merge now :+1:

**Comment by Ma27** [comment]
Backported as 0d38802d66725c7a07ee1f04128c8af50df9b254. Thanks!

**Pull request merged**